nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKudu.java

```diff
@@ -0,0 +1,233 @@
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+public abstract class AbstractKudu extends AbstractProcessor {
+
+    protected static final PropertyDescriptor KUDU_MASTERS = new PropertyDescriptor.Builder()
+            .name("KUDU Masters")
+            .description("List all kudu masters's ip with port (e.g. 7051), comma separated")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("Table Name")
+            .description("The name of the Kudu Table to put data into")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+            .name("record-reader")
+            .displayName("Record Reader")
+            .description("The service for reading records from incoming flow files.")
+            .identifiesControllerService(RecordReaderFactory.class)
+            .required(true)
+            .build();
+
+    protected static final PropertyDescriptor SKIP_HEAD_LINE = new PropertyDescriptor.Builder()
+            .name("Skip head line")
+            .description("Set it to true if your first line is the header line e.g. column names")
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor INSERT_OPERATION = new PropertyDescriptor.Builder()
+            .name("Insert Operation")
+            .description("Specify operation for this processor. Insert-Ignore will ignore duplicated rows")
+            .allowableValues(OperationType.INSERT.toString(), OperationType.INSERT_IGNORE.toString(), OperationType.UPSERT.toString())
+            .defaultValue("Insert")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("A FlowFile is routed to this relationship after it has been successfully stored in Kudu")
+            .build();
+    protected static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("A FlowFile is routed to this relationship if it cannot be sent to Kudu")
+            .build();
+
+    public static final String RECORD_COUNT_ATTR = "record.count";
+
+    protected String kuduMasters;
+    protected String tableName;
+    protected boolean skipHeadLine;
+    protected String operation;
+
+    protected KuduClient kuduClient;
+    protected KuduTable kuduTable;
+
+    @OnScheduled
```

Comment:
I still think we should be doing this `OnEnabled` since Kudu manages the connections for us. If we do it `OnSchedule`, then we end up creating the client **every time** a record is passed through.